resources/js/features/forums/components/EditPostMainRoot/EditPostForm/EditPostForm.tsx

```diff
@@ -0,0 +1,80 @@
+import type { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { BaseAutosizeTextarea } from '@/common/components/+vendor/BaseAutosizeTextarea';
+import { BaseButton } from '@/common/components/+vendor/BaseButton';
+import {
+  BaseFormControl,
+  BaseFormField,
+  BaseFormItem,
+  BaseFormLabel,
+  BaseFormMessage,
+  BaseFormProvider,
+} from '@/common/components/+vendor/BaseForm';
+import { usePageProps } from '@/common/hooks/usePageProps';
+
+import { ShortcodePanel } from '../../ShortcodePanel';
+import { useEditPostForm } from './useEditPostForm';
+
+interface EditPostFormProps {
+  onPreview: (content: string) => void;
+}
+
+export const EditPostForm: FC<EditPostFormProps> = ({ onPreview }) => {
+  const { forumTopicComment } = usePageProps<App.Data.EditForumTopicCommentPageProps>();
+
+  const { t } = useTranslation();
+
+  const { form, mutation, onSubmit } = useEditPostForm(forumTopicComment, {
+    body: forumTopicComment.body,
+  });
+
+  const [body] = form.watch(['body']);
+
+  return (
+    <BaseFormProvider {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)}>
+        <div className="flex flex-col gap-3">
+          <ShortcodePanel />
+
+          <BaseFormField
+            control={form.control}
+            name="body"
+            render={({ field }) => (
+              <BaseFormItem>
+                <BaseFormLabel className="sr-only">{t('Body')}</BaseFormLabel>
+                <BaseFormControl>
+                  <BaseAutosizeTextarea
+                    className="min-h-[308px] p-3"
+                    placeholder="Don't ask for links to copyrighted ROMs. Don't share links to copyrighted ROMs."
+                    minHeight={308}
+                    maxLength={60_000}
+                    {...field}
+                  />
+                </BaseFormControl>
+
+                <BaseFormMessage />
+              </BaseFormItem>
+            )}
+          />
+
+          <div className="flex w-full justify-between gap-3">
+            <span className="text-neutral-400">
+              {t('{{current, number}} / {{max, number}}', { current: body.length, max: 60_000 })}
```

Comment:
I'm not sure how `current` is calculated, but the old number is larger than the new one. I think the old one is counting newlines as two characters.

Reply:
Ah, I didn't even notice. I've extracted our `getStringByteCount()` from the _tall-stack_ stuff and into the React utils directory. It's now used by the React code and still working in the legacy stuff.